File: pset1/mario/more/mario.c
/*CS50x 2021
  Problem Set 1: mario more*/

#include <stdio.h>


int main(void)
{
    int height;
    int get_height();
    void print_pyramid(int n);

    // prompt user for height
    // if height less than 1 or greater than 8.
    do {

        height = get_height();

    } while (height < 1 || height > 8);

    // print hash-pyramid.
    print_pyramid(height);

    return 0;
}


int get_height()
{
    int number;

    printf("Height: ");
    scanf("%i", &number);

    return number;
}


void print_pyramid(int n)
{

    for (int i = 0; i < n; i++) {

        // Print left-side.
        for (int s = n-1; s > i; s--) {
            printf(" ");
        }

        for (int h = 0; h <= i; h++) {
            printf("#");
        }

        printf("  ");

        // Print right-side.
        for (int h = 0; h <= i; h++) {
            printf("#");
        }

        // Print line break.
        printf("\n");
    }
}

File: week6/pset6/mario/less/mario.py
from cs50 import get_int


def get_height(min=1, max=8):
    """Prompt user for height value."""
    while True:
        height = get_int("Height: ")
        if height >= min and height <= max:
            return height


def print_pyramid(n):
    """Print n height of half-pyramid to console."""
    for i in range(1, n+1):
        print(" "*(n-i) + "#"*i)


def main():
    height = get_height()
    print_pyramid(height)


if __name__ == "__main__":
    main()
